### PR TITLE
Balanced joint maximum mean discrepancy for deep transfer learning

### PR DESCRIPTION
**Pull request by mengchuangji**
Balanced joint maximum mean discrepancy for deep transfer learning is a deep learning version of Balanced Distribution Adaptation.
Recent studies have revealed that deep networks can learn transferable features that generalize well to novel tasks with little or unavailable labeled data for domain adaptation. However, justifying which components of the feature representations can reason about original joint distributions using JMMD within the regime of deep architecture remains unclear. We present a new backpropagation algorithm for JMMD called the Balanced Joint Maximum Mean Discrepancy (B-JMMD) to further reduce the domain discrepancy. B-JMMD achieves the effect of balanced distribution adaptation for deep network architecture, and can be treated as an improved version of JMMD’s backpropagation algorithm. The proposed method leverages the importance of marginal and conditional distributions behind multiple domain-specific layers across domains adaptively to get a good match for the joint distributions in a second-order reproducing kernel Hilbert space. The learning of the proposed method can be performed technically by a special form of stochastic gradient descent, in which the gradient is computed by backpropagation with a strategy of balanced distribution adaptation. Theoretical analysis shows that the proposed B-JMMD is superior to JMMD method. Experiments confirm that our method yields state-of-the-art results with standard datasets.

### CODE REVIEW

**Comment by jindongwang** [comment]
@mengchuangji Thanks for your contribution to this project. I noticed that you have committed over 600 files for one single algorithm, which is quite confusing and will put a heavy memory burden for this project. I would like you to confirm that: are all the files necessary for your algorithm? Can you delete some unnecessary files (e.g., files that are automatically generated) to make it clear and concise?

**Comment by mengchuangji** [comment]
Yes, this is a complete Caffe file, all files are required for compilation, and there is no automatically generated file.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jindongwang/transferlearning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月4日(星期二) 中午11:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jindongwang/transferlearning] Balanced joint maximum mean discrepancy for deep transfer learning (PR #299)





 
@mengchuangji Thanks for your contribution to this project. I noticed that you have committed over 600 files for one single algorithm, which is quite confusing and will put a heavy memory burden for this project. I would like you to confirm that: are all the files necessary for your algorithm? Can you delete some unnecessary files (e.g., files that are automatically generated) to make it clear and concise?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jindongwang** [comment]
Although I'm not familiar with Caffe, but I can see that most of the files are Caffe default files, which can be deleted. You only need to add your own codes here and people can download it to combine with Caffe environment to run.

**Comment by mengchuangji** [comment]
Okay, thank you for your suggestion.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jindongwang/transferlearning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月4日(星期二) 中午11:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jindongwang/transferlearning] Balanced joint maximum mean discrepancy for deep transfer learning (PR #299)





 
Although I'm not familiar with Caffe, but I can see that most of the files are Caffe default files, which can be deleted. You only need to add your own codes here and people can download it to combine with Caffe environment to run.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
